pyscriptjs/src/main.ts

```diff
@@ -204,9 +201,34 @@ export class PyScriptApp {
         // XXX: maybe the following calls could be parallelized, instead of
         // await()ing immediately. For now I'm using await to be 100%
         // compatible with the old behavior.
+        logger.info('importing pyscript');
+
+        // Save and load pyscript.py from FS
+        runtime.interpreter.FS.writeFile('pyscript.py', pyscript, { encoding: 'utf8' });
+
+        // inject `define_custom_element` it into the PyScript module scope
+        const pyscript_module = runtime.interpreter.pyimport('pyscript');
+        pyscript_module.define_custom_element = define_custom_element;
+        pyscript_module.PyScript.set_version_info(version);
+        pyscript_module.destroy();
+
+        // TODO: Currently adding the imports for backwards compatibility, we should
+        //       remove it
+        await runtime.run(`
+        from pyscript import *
+        `);
+        logger.warn(`DEPRECATION WARNING: 'micropip', 'Element', 'console', 'document' and several other \
+objects form the pyscript module (with the exception of 'display') will be \
+be removed from the Python global namespace in the following release. \
+To avoid errors in future releases use import from pyscript instead. For instance: \
+from pyscript import micropip, Element, console, document`);
+
```

Comment:
-1 for this: it's a warning which is easy to overlook and most importantly it is shown unconditionally, which will probably confuse people. I'd prefer something along these lines:
```python
class DeprecatedGlobal:
    def __init__(self, name, obj):
        self.name = name
        self.obj = obj
        self.warning_already_shown = False

    def show_warning_maybe(self):
        if self.warning_already_shown:
            return
        showWarning(f"Direct usage of {self.name} is deprecated. Please use pyscript.{self.name} instead")
        self.warning_already_shown = True

    def __getattr__(self, attr):
        self.show_warning_maybe()
        return getattr(self.obj, attr)

    def __call__(self, *args, **kwargs):
        self.show_warning_maybe()
        return self.obj(*args, **kwargs)

Element = DeprecatedGlobal('Element', pyscript.Element)
console = DeprecatedGlobal('console', pyscript.console)
...
```

But I think it's fine to do this in a follow-up request (I'm happy to do it if you want)

Reply:
In the meantime, I found out that the mess is even worse that this, see  https://github.com/pyscript/pyscript/issues/986

Reply:
Meh... I'm -0 because I don't think it's worth the effort and it's not user facing (when user == users of the PyScript application, not devs).

If you want to take a stab at it and wouldn't delay the release even more, I'm not opposed to that

Reply:
> Meh... I'm -0 because I don't think it's worth the effort and it's not user facing (when user == users of the PyScript application, not devs).

I think that this is the core of many disagreements. From my point of view, our users are the developers, and we should try to make the life simpler to them.
It is *their* job to make nice apps for their final users, not ours; our job is to make it as simple as possible to make it so.

> If you want to take a stab at it and wouldn't delay the release even more, I'm not opposed to that

doing a release even if there is a known broken feature "because it's late" doesn't sound like a good idea. We will do a release when it's ready, IMHO.
The history tells us that putting features in a rush "because we need to release" doesn't end well ;)

That said, I'm happy to do it.